unitrade-backend/src/main/java/ca/mcgill/ecse428/unitrade/unitradebackend/controller/CourseRestController.java

```diff
@@ -0,0 +1,83 @@
+package ca.mcgill.ecse428.unitrade.unitradebackend.controller;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import ca.mcgill.ecse428.unitrade.unitradebackend.dto.Request.CourseRequestDto;
+import ca.mcgill.ecse428.unitrade.unitradebackend.dto.Response.CourseResponseDto;
+import ca.mcgill.ecse428.unitrade.unitradebackend.model.Course;
+import ca.mcgill.ecse428.unitrade.unitradebackend.service.CourseService;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+
+@RestController
+@PreAuthorize("hasRole('USER')")
+@ApiResponses(value = {
+    @ApiResponse(responseCode = "201", description = "Course created"),
+    @ApiResponse(responseCode = "400", description = "Invalid input"),
+    @ApiResponse(responseCode = "404", description = "Referenced resource not found"),
+    @ApiResponse(responseCode = "409", description = "Unique constraint violation")
+})
+public class CourseRestController {
+    
+    @Autowired
+    CourseService courseService;
+
+    @ResponseStatus(HttpStatus.CREATED)
+    @PreAuthorize("permitAll()")
```

Comment:
Are we permitting the creation of courses with any authorization?

Reply:
Its a placeholder right now because we dont have any admin classes